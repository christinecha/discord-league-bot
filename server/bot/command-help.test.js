// Start mock server!
require('./index')
const firebase = require('@firebase/rules-unit-testing')
const { discord } = require('../data/util/discord')
const { getCommandsMarkdown } = require('../../shared/getCommandsMarkdown')
const { guild } = require('../test/guild')
const BOT_ID = process.env.BOT_ID

let send, msg

beforeAll(async (done) => {
  await firebase.clearFirestoreData({
    projectId: process.env.GCLOUD_PROJECT,
  })

  done()
})

beforeEach(async (done) => {
  send = jest.fn()
  msg = (userId, content) => ({
    content,
    author: { id: userId },
    guild,
    channel: { send, id: 'test' },
  })

  done()
})

afterEach(async (done) => {
  done()
})

test('@LeagueBot help', async (done) => {
  const commandsMarkdown = getCommandsMarkdown()
  await discord.trigger('message', msg('flips', `<@!${BOT_ID}> help`))

  // Commands
  expect(commandsMarkdown).toMatchSnapshot()

  // Confirmations should be sent.
  expect(send).toHaveBeenCalledWith(
    expect.objectContaining({
      fields: expect.arrayContaining([
        expect.objectContaining({
          name: 'Variables',
        }),
        expect.objectContaining({
          name: 'Commands',
          value: commandsMarkdown,
        }),
      ]),
    })
  )

  done()
})
