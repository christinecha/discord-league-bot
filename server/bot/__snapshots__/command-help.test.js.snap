// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@LeagueBot help 1`] = `
"- \`@LeagueBot cancel <matchId>\` - Cancel a match.
- \`@LeagueBot clear | c <teamSize>\` - Clear the queue for a specific league.
- \`@LeagueBot help | h\` - Show me all the commands.
- \`@LeagueBot leaderboard <[teamSize]>\` - Show me the leaderboard!
- \`@LeagueBot leave | l <[teamSize]>\` - Leave the queue for one or all leagues.
- \`@LeagueBot loss | lose | lost <matchId>\` - Report that you lost this match.
- \`@LeagueBot new | n <teamSize>\` - Start a new league for a specific team size.
- \`@LeagueBot queue | q <teamSize>\` - Join the queue for a specific league.
- \`@LeagueBot status | s <[teamSize]>\` - Show the current queue(s).
- \`@LeagueBot win | won | w <matchId>\` - Report that you won this match."
`;
